tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala

```diff
@@ -1246,6 +1246,58 @@ class ControlSyntaxSuite extends BaseDottySuite {
     )
   }
 
+  test("new-match-case-oneline-align-newline") {
+    val code = """|def fx: String = {
+                  |  x match
+                  |  case 2 =>
+                  |    "ERROR"
+                  |  end match
+                  |}
+                  |""".stripMargin
+    val output = """|def fx: String = {
+                    |  x match {
+                    |    case 2 => "ERROR"
+                    |  }
+                    |  end `match`
+                    |}
+                    |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = Some(output))(
+      Defn.Def(
+        Nil,
+        Term.Name("fx"),
+        Nil,
+        Nil,
+        Some(Type.Name("String")),
+        Term.Block(
+          List(
+            Term.Match(Term.Name("x"), List(Case(Lit.Int(2), None, Lit.String("ERROR")))),
+            Term.EndMarker(Term.Name("match"))
```

Comment:
Any way we can say that it's the keyword actually? Reprinting it might add `` always

Reply:
Yes good catch, it printed match surrounded with ``
The thing is it needs to handle normal name: `end myCusfomFunc` but also keywords `end if`
Now I just changed treeprinting to print back `ternname.value` because printing whole TermName added `` if it was keyword which is probably correct way in rest of the code.

One way would be to keep it as is.
Other way would be to simply keep raw `String` in EndMarker?
Third way would be to introduce sth. like `Either[Term.Name, Token]`

Reply:
Let's leave it as is for now.